receiver/snmpreceiver/goSNMPWrapper.go

```diff
@@ -0,0 +1,231 @@
+	// packets. And if the host is regularly disappearing and reappearing, you won't
+	// know if you've only done a Connect().
+	//
+	// For historical reasons (ie this is part of the public API), the method won't
+	// be renamed.
+	Connect() error
+
+	// Close closes the connection
+	Close() error
+
+	// Get sends an SNMP GET request
+	Get(oids []string) (result *gosnmp.SnmpPacket, err error)
+
+	// Walk retrieves a subtree of values using GETNEXT - a request is made for each
+	// value, unlike BulkWalk which does this operation in batches. As the tree is
+	// walked walkFn is called for each new value. The function immediately returns
+	// an error if either there is an underlaying SNMP error (e.g. GetNext fails),
+	// or if walkFn returns an error.
+	Walk(rootOid string, walkFn gosnmp.WalkFunc) error
+
+	// BulkWalk retrieves a subtree of values using GETBULK. As the tree is
+	// walked walkFn is called for each new value. The function immediately returns
+	// an error if either there is an underlaying SNMP error (e.g. GetBulk fails),
+	// or if walkFn returns an error.
+	BulkWalk(rootOid string, walkFn gosnmp.WalkFunc) error
+
+	// GetTransport gets the Transport
+	GetTransport() string
+
+	// SetTransport sets the Transport
+	SetTransport(transport string)
+
+	// GetTarget gets the Target
+	GetTarget() string
+
+	// SetTarget sets the Target
+	SetTarget(target string)
+
+	// GetPort gets the Port
+	GetPort() uint16
+
+	// SetPort sets the Port
+	SetPort(port uint16)
+
+	// GetCommunity gets the Community
+	GetCommunity() string
+
+	// SetCommunity sets the Community
+	SetCommunity(community string)
+
+	// GetVersion gets the Version
+	GetVersion() gosnmp.SnmpVersion
+
+	// SetVersion sets the Version
+	SetVersion(version gosnmp.SnmpVersion)
+
+	// GetTimeout gets the Timeout
+	GetTimeout() time.Duration
+
+	// SetTimeout sets the Timeout
+	SetTimeout(timeout time.Duration)
+
+	// GetMaxOids gets the MaxOids
+	GetMaxOids() int
+
+	// SetMaxOids sets the MaxOids
+	SetMaxOids(maxOids int)
+
+	// GetMsgFlags gets the MsgFlags
+	GetMsgFlags() gosnmp.SnmpV3MsgFlags
+
+	// SetMsgFlags sets the MsgFlags
+	SetMsgFlags(msgFlags gosnmp.SnmpV3MsgFlags)
+
+	// GetSecurityModel gets the SecurityModel
+	GetSecurityModel() gosnmp.SnmpV3SecurityModel
+
+	// SetSecurityModel sets the SecurityModel
+	SetSecurityModel(securityModel gosnmp.SnmpV3SecurityModel)
+
+	// GetSecurityParameters gets the SecurityParameters
+	GetSecurityParameters() gosnmp.SnmpV3SecurityParameters
+
+	// SetSecurityParameters sets the SecurityParameters
+	SetSecurityParameters(securityParameters gosnmp.SnmpV3SecurityParameters)
+}
+
+// otelGoSNMPWrapper is a wrapper around gosnmp
+type otelGoSNMPWrapper struct {
+	gosnmp.GoSNMP
+}
+
+// newGoSNMPWrapper creates a new goSNMPWrapper using gosnmp
+func newGoSNMPWrapper() goSNMPWrapper {
+	return &otelGoSNMPWrapper{
+		gosnmp.GoSNMP{
+			MaxOids: gosnmp.Default.MaxOids,
```

Comment:
Is there a case where the MaxOids could cause an issue for users? Not sure what this limit enforces.